### PR TITLE
feat: [CDS-88614]: Added yaml example for notification rules v1

### PR DESCRIPTION
**Pull request by shalini-agr**

### CODE REVIEW

File: v1/examples/pipeline/pipeline_with_notification_rules.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        - start
+        - end
+        - success
+        - fail
+      stage_events:
+        - type: start
+          ids:
+            - all
+        - type: success
+          ids:
```
Comment (archit-harness) [inline v1/examples/pipeline/pipeline_with_notification_rules.yaml:27]: we should convert this to ref

File: v1/examples/pipeline/pipeline_with_notification_rules.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            - id1
+        - type: fail
+          ids:
```
Comment (archit-harness) [inline v1/examples/pipeline/pipeline_with_notification_rules.yaml:30]: same as above

File: v1/examples/pipeline/pipeline_with_notification_rules.yaml (hunk 1 above)
Comment (archit-harness) [reply v1/examples/pipeline/pipeline_with_notification_rules.yaml:27]: refs

File: v1/examples/pipeline/pipeline_with_notification_rules.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        steps:
+          - type: plugin
+            spec:
+              uses: http
+              with:
+                url: <+ stages.nested.outputs.url>
+                method: GET
+  timeout: 10m
+  notification_rules:
+    n1:
```
[the remark this answers is not in the document]
Comment (shalini-agr) [reply v1/examples/pipeline/pipeline_with_notification_rules.yaml:16]: i thought we could have it as a a map with identifier as string
just like variables

File: v1/examples/pipeline/pipeline_with_notification_rules.yaml (hunk 2 above)
Comment (shalini-agr) [reply v1/examples/pipeline/pipeline_with_notification_rules.yaml]: done

File: v1/examples/pipeline/pipeline_with_notification_rules.yaml (hunk 1 above)
Comment (shalini-agr) [reply v1/examples/pipeline/pipeline_with_notification_rules.yaml]: done

**Review by SudarshanPurohit — APPROVED** [review]
Overall looks good. Please make sure to capture the Header support for Webhooks (https://harness.atlassian.net/browse/CDS-84456) in the spec as we implement.

**Comment by archit-harness** [comment]
@SudarshanPurohit the header spec is still under impl? or its schema changes were done?

**Review by archit-harness — DISMISSED** [review]

**Review by SudarshanPurohit — APPROVED** [review]
All comments have been resolved. Approving this now.

**Review by archit-harness — APPROVED** [review]

**Pull request merged**